libdnf/goal/Goal.cpp

```diff
@@ -1147,14 +1148,11 @@ Goal::Impl::allowUninstallAllButProtected(Queue *job, DnfGoalActions flags)
     } else
         map_grow(protectedPkgs->getMap(), pool->nsolvables);
 
-    Id kernel = dnf_sack_running_kernel(sack);
-    if (kernel > 0)
-        protectedPkgs->set(kernel);
-
     if (DNF_ALLOW_UNINSTALL & flags)
         for (Id id = 1; id < pool->nsolvables; ++id) {
             Solvable *s = pool_id2solvable(pool, id);
             if (pool->installed == s->repo && !protectedPkgs->has(id) &&
+                !isProtectedRunningKernel(id) &&
```

Comment:
Performance regression. From constant complexity to linear. dnf_sack_ruinning_kernel() was called once before the PR. But it called (via isProtectedRunningKernel) for each installed non protected package now.

Reply:
I changed it so that the ``dnf_sack_ruinning_kernel()`` is called only once in ``constructJob()`` and the id is stored in a variable. Is this approach ok?

Reply:
I hope, yes.

Reply:
@jrohel Why you mentioned dnf_sack_ruinning_kernel()  as a performance regression? It detects running kernel and then it is used as a getter.

Reply:
What about to remove isProtectedRunningKernel(), runningKernelID or create new function that will return protected kernel if it has to be protected.

```
Id
Goal::Impl::ProtectedRunningKernel()
{
    return protect_running_kernel ? dnf_sack_running_kernel(sack) : 0;
}
```


```
    Id protected_kernel = ProtectedRunningKernel()
    if (DNF_ALLOW_UNINSTALL & flags)
        for (Id id = 1; id < pool->nsolvables; ++id) {
            Solvable *s = pool_id2solvable(pool, id);
            if (pool->installed == s->repo && !protectedPkgs->has(id) &&
                id !=  protected_kernel &&
```